srs-api/src/main/resources/sql/schema_triggers.sql

```diff
@@ -14,3 +14,83 @@ BEGIN
         add_course_credit(:NEW.COURSE#);
     END IF;
 END;^
+
+/*
+    8.(8 points) Write triggers to add tuples to the Logs table automatically whenever a student is deleted
+    from the Students table, or when a student is successfully enrolled into or dropped from a class (i.e.,
+    when a tuple is inserted into or deleted from the G_Enrollments table). For a logs record for
+    G_Enrollments, the key value is the concatenation of the B# value, a comma, and the classid value.
+*/
+
+CREATE OR REPLACE
+TRIGGER student
+BEFORE
+DELETE
+ON
+    STUDENTS
+FOR EACH ROW
+BEGIN
+DELETE
+FROM
+    G_ENROLLMENTS ge
+WHERE
+        ge."G_B#" =:OLD.B#;
+
+INSERT INTO LOGS
+VALUES
+    (logseq.NEXTVAL,
+     (SELECT USER FROM dual),
+     CURRENT_TIMESTAMP,
+     'STUDENTS',
+     'DELETE',
+     :OLD.B#);
+END;
+/
+show errors
+
+CREATE OR REPLACE
+TRIGGER enrollment
+BEFORE
+INSERT
+    OR
+DELETE
+ON
+    G_ENROLLMENTS
+FOR EACH ROW
+BEGIN
+    IF INSERTING THEN
+        INSERT INTO LOGS
+            VALUES
+                (logseq.NEXTVAL,
+                (SELECT USER FROM dual),
+                CURRENT_TIMESTAMP,
+                'G_ENROLLMENTS',
+                'INSERT',
+                TO_CHAR(:NEW.G_B#) || ',' || TO_CHAR (:NEW.CLASSID));
+UPDATE
+    CLASSES c
+SET
+    c.CLASS_SIZE = c.CLASS_SIZE  + 1
+WHERE
+        c.CLASSID = :NEW.CLASSID;
+
+ELSIF DELETING THEN
+        INSERT INTO LOGS
+            VALUES
+                (logseq.NEXTVAL,
+                (SELECT USER FROM dual),
+                CURRENT_TIMESTAMP,
+                'G_ENROLLMENTS',
+                'DELETE',
+                TO_CHAR (:OLD.G_B#) || ',' || TO_CHAR (:OLD.CLASSID));
+UPDATE
+    CLASSES c
+SET
+    c.CLASS_SIZE = c.CLASS_SIZE  - 1
+WHERE
+        c.CLASSID = :OLD.CLASSID;
+END IF;
+END;
```

Comment:
all procedural block statements need to end with ";^"

Reply:
when executed adding ";^" is interpreted as "; ;" giving sql syntax error